### PR TITLE
Adds Screenshot testing with Roborazzi

### PR DESCRIPTION
**Pull request by juliewww**

### CODE REVIEW

File: .github/workflows/Build.yaml [hunk 1]
```diff
@@ -115,8 +115,64 @@ jobs:
         with:
           apk-path: ${{ steps.apk-path.outputs.foss_path }}
 
+  test:
+    runs-on: ubuntu-latest
```
Comment (lihenggui) [inline .github/workflows/Build.yaml:119]:
```suggestion
    runs-on: self-hosted
```

File: .github/workflows/Build.yaml [hunk 2]
```diff
@@ -115,8 +115,64 @@ jobs:
...
+          distribution: 'zulu'
+          java-version: 17
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2.8.0
+
+      - name: Run all local screenshot tests (Roborazzi)
+        id: screenshotsverify
+        continue-on-error: true
+        run: ./gradlew verifyRoborazziDemoDebug
```
Comment (lihenggui) [inline .github/workflows/Build.yaml:148]:
```suggestion
        run: ./gradlew verifyRoborazziFossDebug
```

File: .github/workflows/Build.yaml [hunk 3]
```diff
@@ -115,8 +115,64 @@ jobs:
...
+        if: steps.screenshotsverify.outcome == 'failure' && github.event.pull_request.head.repo.full_name != github.repository
+        run: |
+          echo "::error::Screenshot tests failed, please create a PR in your fork first." && exit 1
+
+      # Runs if previous job failed
+      - name: Generate new screenshots if verification failed and it's a PR
+        id: screenshotsrecord
+        if: steps.screenshotsverify.outcome == 'failure' && github.event_name == 'pull_request'
+        run: |
+          ./gradlew recordRoborazziDemoDebug
```
Comment (lihenggui) [inline .github/workflows/Build.yaml:162]:
```suggestion
          ./gradlew recordRoborazziFossDebug
```

File: .github/workflows/Build.yaml [hunk 4]
```diff
@@ -115,8 +115,64 @@ jobs:
...
+        with:
+          file_pattern: '*/*.png'
+          disable_globbing: true
+          commit_message: "🤖 Updates screenshots"
+
+      # Run local tests after screenshot tests to avoid wrong UP-TO-DATE. TODO: Ignore screenshots.
       - name: Run local tests
-        run: ./gradlew testfossDebug testMarketDebug
+        if: always()
+        run: ./gradlew testDemoDebug testProdDebug
```
Comment (lihenggui) [inline .github/workflows/Build.yaml:175]:
```suggestion
        run: ./gradlew testFossDebug testMarketDebug
```

File: README.md [hunk 5]
```diff
@@ -52,6 +52,13 @@ Starting from Android O, if we install a Test-Only application, users could use
 
 Tutorial for modifying APKs (Chinese Only) [[实验性功能] [开发者向]如何免Root控制应用程序组件](https://github.com/lihenggui/blocker/wiki/%5B%E5%AE%9E%E9%AA%8C%E6%80%A7%E5%8A%9F%E8%83%BD%5D-%5B%E5%BC%80%E5%8F%91%E8%80%85%E5%90%91%5D%E5%A6%82%E4%BD%95%E5%85%8DRoot%E6%8E%A7%E5%88%B6%E5%BA%94%E7%94%A8%E7%A8%8B%E5%BA%8F%E7%BB%84%E4%BB%B6)
 
+## Screenshot tests
+
+**Now In Android** uses [Roborazzi](https://github.com/takahirom/roborazzi) to do screenshot tests
```
Comment (lihenggui) [inline README.md:57]: You should check the content you pasted before committing to this repo.

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -164,6 +169,7 @@ android-gradlePlugin = { group = "com.android.tools.build", name = "gradle", ver
 kotlin-gradlePlugin = { group = "org.jetbrains.kotlin", name = "kotlin-gradle-plugin", version.ref = "kotlin" }
 ksp-gradlePlugin = { group = "com.google.devtools.ksp", name = "com.google.devtools.ksp.gradle.plugin", version.ref = "ksp" }
 material = { group = "com.google.android.material", name = "material", version.ref = "material" }
+work-testing = { group = "androidx.work", name = "work-testing", version = "2.8.1" }
```
Comment (lihenggui) [inline gradle/libs.versions.toml:172]: Does this dependency get used?

Comment (juliewww) [reply gradle/libs.versions.toml:172]: It seems like not, delete it.

**Review by lihenggui — APPROVED** [review]

**Pull request merged**